File: core/src/main/java/com/jinkyumpark/bookitout/search/apiResponse/mille/ApiMillieBook.java
package com.jinkyumpark.bookitout.search.apiResponse.mille;

import com.jinkyumpark.bookitout.search.provider.SubscriptionProvider;
import com.jinkyumpark.bookitout.search.response.searchResult.SubscriptionSearchResult;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor @AllArgsConstructor
@Getter
public class ApiMillieBook {

    private String book_seq;
    private String content_name;
    private String category;
    private String category2;
    private String category_seq;
    private String category_seq2;
    private String content_thumb_url;
    private String subtitle;
    private String view_percent;
    private String view_time;
    private String author;
    private String content_code;
    private String audiobook_volume;
    private String audio_category_icon_value;
    private String audio_reader_icon_value;
    private String reader_name;
    private String reader_job;
    private String is_service;
    private String is_free;
    private String is_adult;
    private String is_b2b;
    private String is_series;
    private String is_series_complete;
    private String book_kind_code;
    private String series_count;
    private String series_group_name;
    private String series_type;
    private String coming_soon_date;
    private String is_ebook_rent;
    private String book_brand;
    private String content_take_count;
    private String content_review_count;
    private String ebook_published_at;

    public SubscriptionSearchResult toSubscriptionSearchResult() {
        return SubscriptionSearchResult.builder()
                .title(this.content_name)
                .author(this.author)
                .cover(this.content_thumb_url)
                .link("https://www.millie.co.kr/v3/bookDetail/" + this.book_seq)
                .provider(SubscriptionProvider.MILLIE)
                .build();
    }
}

File: search/src/main/kotlin/com/jinkyumpark/search/apiResponse/millie/ApiMillieBook.kt
package com.jinkyumpark.search.apiResponse.millie

import com.jinkyumpark.search.common.SearchProvider
import com.jinkyumpark.search.common.SearchResult

data class ApiMillieBook(
    val book_seq: String?,
    val content_name: String,
    val category: String?,
    val category2: String?,
    val category_seq: String?,
    val category_seq2: String?,
    val content_thumb_url: String?,
    val subtitle: String?,
    val view_percent: String?,
    val view_time: String?,
    val author: String,
    val content_code: String?,
    val audiobook_volume: String?,
    val audio_category_icon_value: String?,
    val audio_reader_icon_value: String?,
    val reader_name: String?,
    val reader_job: String?,
    val is_service: String?,
    val is_free: String?,
    val is_adult: String?,
    val is_b2b: String?,
    val is_series: String?,
    val is_series_complete: String?,
    val book_kind_code: String?,
    val series_count: String?,
    val series_group_name: String?,
    val series_type: String?,
    val coming_soon_date: String?,
    val is_ebook_rent: String?,
    val book_brand: String?,
    val content_take_count: String?,
    val content_review_count: String?,
    val ebook_published_at: String?,
) {
    fun toBookSearchResult(): SearchResult {
        return SearchResult(
            title = content_name,
            author = author,
            cover = content_thumb_url,
            link = "https://www.millie.co.kr/v3/bookDetail/$book_seq",
            searchProvider = SearchProvider.MILLIE_SUBSCRIPTION,
            isbn = null,
        )
    }

}